src/tyr/route_serializer_osrm.cc

```diff
@@ -489,6 +489,12 @@ json::ArrayPtr intersections(const valhalla::DirectionsLeg::Maneuver& maneuver,
       if (curr_edge->use() == TripLeg::Use::TripLeg_Use_kFerryUse) {
         classes.push_back("ferry");
       }
+      if (curr_edge->use() == TripLeg::Use::TripLeg_Use_kRestAreaUse) {
+        classes.push_back("rest_area");
+      }
+      if (curr_edge->use() == TripLeg::Use::TripLeg_Use_kServiceAreaUse) {
+        classes.push_back("service_area");
```

Comment:
Rather than add to `classes`, we should put these in a new property on the intersection object.